Correction of NRE, Paramatrized attenuation and Reduced the current attenuation of Jukebox

JesterX666 opened the pull request:
Correction of NRE, Paramatrized attenuation and Reduced the current attenuation of Jukebox.

I could have use a StringBuilder instead of string interpolation, but I find it "clearer" the way I did it.

NRE Detail:

NullReferenceException: Object reference not set to an instance of an object
Assets.Scripts.Messages.Server.SoundMessages.PlaySoundMessage.ToString () (at /github/workspace/UnityProject/Assets/Scripts/Messages/Server/SoundMessages/PlaySoundMessage.cs:154)
Assets.Scripts.Messages.Server.SoundMessages.PlaySoundMessage.Process () (at /github/workspace/UnityProject/Assets/Scripts/Messages/Server/SoundMessages/PlaySoundMessage.cs:28)
GameMessageBase.Process (Mirror.NetworkConnection sentBy) (at /github/workspace/UnityProject/Assets/Scripts/Messages/GameMessageBase.cs:25)
GameMessageBase.PreProcess (Mirror.NetworkConnection sentBy, GameMessageBase b) (at /github/workspace/UnityProject/Assets/Scripts/Messages/GameMessageBase.cs:18)
Mirror.MessagePacker+<>c__DisplayClass6_0`2[T,C].<MessageHandler>b__0 (Mirror.NetworkConnection conn, Mirror.NetworkReader reader, System.Int32 channelId) (at /github/workspace/UnityProject/Assets/Mirror/Runtime/MessagePacker.cs:146)
Mirror.NetworkConnection.InvokeHandler (System.Int32 msgType, Mirror.NetworkReader reader, System.Int32 channelId) (at /github/workspace/UnityProject/Assets/Mirror/Runtime/NetworkConnection.cs:219)
Mirror.NetworkConnection.TransportReceive (System.ArraySegment`1[T] buffer, System.Int32 channelId) (at /github/workspace/UnityProject/Assets/Mirror/Runtime/NetworkConnection.cs:272)
Mirror.ULocalConnectionToServer.Update () (at /github/workspace/UnityProject/Assets/Mirror/Runtime/LocalConnections.cs:110)
Mirror.NetworkClient.Update () (at /github/workspace/UnityProject/Assets/Mirror/Runtime/NetworkClient.cs:259)
Mirror.NetworkManager.LateUpdate () (at /github/workspace/UnityProject/Assets/Mirror/Runtime/NetworkManager.cs:255)

JesterX666 committed:
Correction of a NRE + Sound Distance parameters for the Jukebox

Correction of a NRE + Sound Distance parameters for the Jukebox

NRE details:

NullReferenceException: Object reference not set to an instance of an object
Assets.Scripts.Messages.Server.SoundMessages.PlaySoundMessage.ToString () (at /github/workspace/UnityProject/Assets/Scripts/Messages/Server/SoundMessages/PlaySoundMessage.cs:154)
Assets.Scripts.Messages.Server.SoundMessages.PlaySoundMessage.Process () (at /github/workspace/UnityProject/Assets/Scripts/Messages/Server/SoundMessages/PlaySoundMessage.cs:28)
GameMessageBase.Process (Mirror.NetworkConnection sentBy) (at /github/workspace/UnityProject/Assets/Scripts/Messages/GameMessageBase.cs:25)
GameMessageBase.PreProcess (Mirror.NetworkConnection sentBy, GameMessageBase b) (at /github/workspace/UnityProject/Assets/Scripts/Messages/GameMessageBase.cs:18)
Mirror.MessagePacker+<>c__DisplayClass6_0`2[T,C].<MessageHandler>b__0 (Mirror.NetworkConnection conn, Mirror.NetworkReader reader, System.Int32 channelId) (at /github/workspace/UnityProject/Assets/Mirror/Runtime/MessagePacker.cs:146)
Mirror.NetworkConnection.InvokeHandler (System.Int32 msgType, Mirror.NetworkReader reader, System.Int32 channelId) (at /github/workspace/UnityProject/Assets/Mirror/Runtime/NetworkConnection.cs:219)
Mirror.NetworkConnection.TransportReceive (System.ArraySegment`1[T] buffer, System.Int32 channelId) (at /github/workspace/UnityProject/Assets/Mirror/Runtime/NetworkConnection.cs:272)
Mirror.ULocalConnectionToServer.Update () (at /github/workspace/UnityProject/Assets/Mirror/Runtime/LocalConnections.cs:110)
Mirror.NetworkClient.Update () (at /github/workspace/UnityProject/Assets/Mirror/Runtime/NetworkClient.cs:259)
Mirror.NetworkManager.LateUpdate () (at /github/workspace/UnityProject/Assets/Mirror/Runtime/NetworkManager.cs:255)

JesterX666 committed:
Augmented the attenuation (reduced the max distance)

NoooneyDude commented on UnityProject/Assets/Scripts/Machines/Jukebox.cs:
```diff
@@ -34,6 +34,12 @@ public class Jukebox : NetworkBehaviour, IAPCPowered
+	private float MinSoundDistance;
+
+	[SerializeField]
+	private float MaxSoundDistance;
```
Could you assign any non-public `[SerializeField]`s to some value, like `default` or `null`. Just helps reduce compiler warnings. Thanks.

JesterX666 replied:
Done

JesterX666 committed:
Default values

JesterX666 committed:
Merge branch 'develop' into Jukebox_New

JesterX666 committed:
Adjustements to have a decent sound attenuation.

Allow to specify which type of volume rolloff we want, and allow a nice ease-in, ease-out sound curve.

JesterX666 committed:
Update Jukebox.prefab

JesterX666 committed:
Merge branch 'Jukebox_New' of https://github.com/JesterX666/unitystation into Jukebox_New
